vita3k/main.cpp

```diff
@@ -117,66 +117,77 @@ int main(int argc, char *argv[]) {
 #if DISCORD_RPC
             discord::update_init_status(host.cfg.discord_rich_presence, &discord_rich_presence_old);
 #endif
             gui::draw_ui(gui, host);
             gui::draw_game_selector(gui, host);
 
-            gui::draw_end(gui, host.window.get());
+            // TODO: Clean this, ie. make load_app overloads called depending on run type
+            if (!gui.game_selector.selected_title_id.empty()) {
+                vpk_path_wide = string_utils::utf_to_wide(gui.game_selector.selected_title_id);
+                host.run_type = app::AppRunType::Extracted;
+            }
         } else {
-            return QuitRequested;
-        }
-
-        // TODO: Clean this, ie. make load_app overloads called depending on run type
-        if (!gui.game_selector.selected_title_id.empty()) {
-            vpk_path_wide = string_utils::utf_to_wide(gui.game_selector.selected_title_id);
-            run_type = app::AppRunType::Extracted;
-        }
-    }
-
-    Ptr<const void> entry_point;
-    if (const auto err = load_app(entry_point, host, gui, vpk_path_wide, run_type) != Success)
-        return err;
-    if (const auto err = run_app(host, entry_point) != Success)
-        return err;
-
-    gui.imgui_state->do_clear_screen = false;
+            if (host.kernel.threads.empty()) {
+                auto pos = 0;
+                for (auto &aloc : host.mem.allocated_pages) {
+                    if (aloc != 0)
+                        LOG_DEBUG("aloc: {}, pos: {}", aloc, pos);
+                    ++pos;
+                }
+                for (auto &gen_name : host.mem.generation_names) {
+                    if (!gen_name.second.empty())
+                        LOG_DEBUG("gen_name: {} & {}, pos: {}", gen_name.first, gen_name.second, pos);
+                    ++pos;
+                }
+
+                //LOG_DEBUG("page size: {}", host.mem.page_size);
+                Ptr<const void> entry_point;
+                if (const auto err = load_app(entry_point, host, gui, vpk_path_wide) != Success)
+                    return err;
+                if (const auto err = run_app(host, entry_point) != Success)
+                    return err;
+
+                gui.imgui_state->do_clear_screen = false;
+
+                if (!gl_renderer.init(host.base_path))
+                    return RendererInitFailed;
+            }
 
-    app::gl_screen_renderer gl_renderer;
+            // Driver acto!
+            renderer::process_batches(*host.renderer.get(), host.renderer->features, host.mem, host.cfg, host.base_path.c_str(),
+                host.io.title_id.c_str());
 
-    if (!gl_renderer.init(host.base_path))
-        return RendererInitFailed;
+            gl_renderer.render(host);
 
-    while (handle_events(host, gui)) {
-        // Driver acto!
-        renderer::process_batches(*host.renderer.get(), host.renderer->features, host.mem, host.cfg, host.base_path.c_str(),
-            host.io.title_id.c_str());
+            // Calculate FPS
```

Comment:
This comment is unnecessary.

Reply:
this comment ,is not me, i have unctouched it, is master code

Reply:
bruh leave it there